Kernel/Syscalls/prctl.cpp

```diff
@@ -12,111 +12,113 @@ namespace Kernel {
 ErrorOr<FlatPtr> Process::sys$prctl(int option, FlatPtr arg1, FlatPtr arg2, FlatPtr arg3)
 {
     VERIFY_NO_PROCESS_BIG_LOCK(this);
-    return with_mutable_protected_data([&](auto& protected_data) -> ErrorOr<FlatPtr> {
-        switch (option) {
-        case PR_GET_DUMPABLE:
+    switch (option) {
+    case PR_GET_DUMPABLE:
+        return with_protected_data([&](auto& protected_data) {
```

Comment:
Looking at the `SpinLockProtectedBase`, exposing it's internal RAII-like `Locker` with a nice name
like `acquire()` or `access()` may make this code a bit nicer (and probably all other users that blew up in size

Reply:
Weren't many places then entering the scope again, or was I misreading things

Reply:
Ah checker locally, where I can hide whitespace changes, and yeeh only two places where actually using that lock